packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/scheduleWalkthrough.ts

```diff
@@ -0,0 +1,170 @@
+    }
+  }
+  return scheduleParams;
+}
+
+async function cronServiceWalkthrough(context: any) {
+  let cloudwatchRule;
+  // resource questions for setting cron
+  const intervalQuestion = {
+    type: 'list',
+    name: 'interval',
+    message: 'At which interval should the function be invoked:',
+    choices: [MINUTES, HOURLY, DAILY, WEEKLY, MONTHLY, YEARLY, CUSTOM],
+  };
+  const intervalAnswer = await inquirer.prompt([intervalQuestion]);
+  switch (intervalAnswer.interval) {
+    case MINUTES: {
+      cloudwatchRule = minuteHelper(context);
+      break;
+    }
+    case HOURLY: {
+      cloudwatchRule = hourHelper(context);
+      break;
+    }
+    case DAILY: {
+      var exp = new CronBuilder();
+      exp = await timeHelper(exp);
+      cloudwatchRule = exp.build();
+      cloudwatchRule = 'cron(' + replaceAt(cloudwatchRule, cloudwatchRule.lastIndexOf('*'), '?') + ' ' + '*' + ')';
+      break;
+    }
+    case WEEKLY: {
+      var exp1 = new CronBuilder();
+      exp1 = await weekHelper(exp1);
+      exp1 = await timeHelper(exp1);
+      cloudwatchRule = exp1.build();
+      cloudwatchRule = 'cron(' + cloudwatchRule + ' ' + '*' + ')';
+      break;
+    }
+    case MONTHLY: {
+      var exp2 = new CronBuilder();
+      exp2 = await monthHelper(exp2, context);
+      exp2 = await timeHelper(exp2);
+      cloudwatchRule = exp2.build();
+      cloudwatchRule = 'cron(' + replaceAt(cloudwatchRule, cloudwatchRule.lastIndexOf('*'), '?') + ' ' + '*' + ')';
+      break;
+    }
+    case YEARLY: {
+      var exp3 = new CronBuilder();
+      exp3 = await yearHelper(exp3, context);
+      exp3 = await timeHelper(exp3);
+      cloudwatchRule = exp3.build();
+      cloudwatchRule = 'cron(' + replaceAt(cloudwatchRule, cloudwatchRule.lastIndexOf('*'), '?') + ' ' + '*' + ')';
+
+      break;
+    }
+    case CUSTOM: {
+      const customRuleQuestion = {
+        type: 'input',
+        name: 'customRule',
+        message: 'Custom Schedule expression(Learn more: https://amzn.to/3akXtJF)',
+        validate: ValidCronExpression({
+          onErrorMsg: 'Enter a valid Schedule Expression (Learn more: https://amzn.to/3akXtJF)',
+        }),
+      };
+      const customRuleAnswer = await inquirer.prompt([customRuleQuestion]);
+      cloudwatchRule = 'cron(' + customRuleAnswer.customRule + ')';
+
+      break;
+    }
+  }
+  return cloudwatchRule;
+}
+
+function replaceAt(string, index, replace) {
+  return string.substring(0, index) + replace + string.substring(index + 1);
+}
+
+module.exports = {
+  scheduleWalkthrough,
+  cronServiceWalkthrough,
+  isValidCronExpression,
+};
+
+function ValidCronExpression(validation) {
+  return input => {
+    return isValidCronExpression(input) ? true : validation.onErrorMsg;
+  };
+}
+
+export function isValidCronExpression(cronExpression: string): boolean {
+  try {
+    new CronExpression(cronExpression);
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
```

Comment:
since your error message looks to always be the one specified on line 135, why not have isValidCronExpression return that error message on false and use that function reference directly in the prompt validator?

Reply:
I am also using the isValidCronExpression function for unit testing , thats why I kept two functions, one for input validation and other for unit testing for cron expressions

Reply:
don't export an internal function for testing, that's leaking. Export the function above and - why is is PascalCased? - and in the test just use the result. Basically the CronExpression class could have a static method that checks validity and encapsulates this, even better it would be nicer if parsing would not throw error, but instead return a named tuple `(value: CronExpression, error: string)` so no try catches would be needed and also the value would be available.